lib/matplotlib/rcsetup.py

```diff
@@ -931,13 +931,29 @@ def _validate_linestyle(ls):
                          "sequence nor a valid string.".format(ls))
 
     # Look for an on-off ink sequence (in points) *of even length*.
-    # Offset is set to None.
+    # Offset is set to None if not present.
     try:
+        offset = None
+        # Look first for the case (offset, on-off seq). Take also care of the
+        # historically supported corner-case of an on-off seq like ["1", "2"].
+        if (len(ls) == 2 and iterable(ls[1]) and
```

Comment:
With `len(ls) == 2` you implicitly support other than `tuple` containers (e.g. `[None, [1.23, 456]]`), if it is an expected behavior there is not test for it.

Reply:
Well it is True that I did not realized how explicit the docstring was about “dash tuple” for `(offset, on-off seq.)`. :sheep:. 

*I guess that it means that the validator is actually a bit too loose and should be changed to accept only tuples for the case `(offset, on-off seq.)`... Or do we prefer to support non-tuple cases as well?*

Reply:
In general we try to as forgiving as possible on inputs, I do not see a reason to be strict here.

Reply:
@tacaswell Well then should we change the docstrings that explicitly state "dash *tuple*" for a custom line style argument (more of less everywhere :/)? Or is the currently implicit behavior fine? If we want to keep the valid pattern loose, I will at least add a few more tests with other than tuple containers, as @Kojoley suggested.